### PR TITLE
Payment Processor Interface (missing USD/ETH conversion)

### PR DESCRIPTION
**Pull request by end-r**

### CODE REVIEW

File: contracts/platform/__tests__/pay/vendor.test.ts [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+    let vendor: TestVendor;
+    let sku = keccak256('0x00');
+    let nonce = 0;
+
+    beforeEach(async () => {
+        nonce = 0;
+        pay = await new PayFactory(user).deploy();
+        vendor = await new TestVendorFactory(user).deploy(pay.address);
+    });
+
```
Comment (kermankohli) [inline contracts/platform/__tests__/pay/vendor.test.ts]: Could you please refactor to use the subject pattern as per previous test cases please? Helps with readability and ensuring the standard of our tests is consistent with the existing code base.

File: contracts/platform/__tests__/pay/vendor.test.ts [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+    function getETHOrder(sku: string, qty: number, amount: number) {
```
Comment (kermankohli) [inline contracts/platform/__tests__/pay/vendor.test.ts]: Would it be possible to move these helper functions outside the test scope and into a helper file or just at the bottom?

File: contracts/platform/contracts/pay/Pay.sol [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+        bytes32 recoveryHash = keccak256(abi.encodePacked("\x19Ethereum Signed Message:\n32", sigHash));
+        return ecrecover(recoveryHash, receipt.v, receipt.r, receipt.s);
+    }
+
+    function setSignerLimit(address signer, uint256 usdCentsLimit) public onlyOwner {
+        signerLimits[signer].limit = usdCentsLimit;
+        emit SignerLimitChanged(signer, usdCentsLimit);
+    }
+
+    function setSellerApproval(address seller, bytes32[] memory skus, bool approved) public onlyOwner {
```
Comment (kermankohli) [inline contracts/platform/contracts/pay/Pay.sol]: Formatting please - anything with more than 2 parameters should be split across multiple lines for readability :)

File: contracts/platform/contracts/token/BatchTransfer.sol [hunk 4]
```diff
@@ -0,0 +1,19 @@
+pragma solidity 0.5.11;
+
+interface BatchTransfer {
```
Comment (kermankohli) [inline contracts/platform/contracts/token/BatchTransfer.sol]: Are there any instances where you just use BatchTransfer and not ListTransfer? If not, then should merge these into a single class.

File: contracts/platform/__tests__/pay/processor.test.ts [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+  });
+
+  describe('#setSignerLimit', () => {
+
+    let processor: Processor;
+
+    beforeEach(async () => {
+        processor = await new ProcessorFactory(user).deploy();
+    });
+
```
Comment (kermankohli) [inline contracts/platform/__tests__/pay/processor.test.ts]: Test case for if you call as someone other than the owner

File: contracts/platform/contracts/token/BatchTransfer.sol (hunk 4 above)
Comment (end-r) [reply contracts/platform/contracts/token/BatchTransfer.sol:3]: This is for scenarios where different external contracts implement one or other of these, but not both.

File: contracts/platform/__tests__/pay/vendor.test.ts (hunk 2 above)
Comment (end-r) [reply contracts/platform/__tests__/pay/vendor.test.ts]: Should be done in the latest version!

File: contracts/platform/contracts/pay/Pay.sol (hunk 3 above)
Comment (end-r) [reply contracts/platform/contracts/pay/Pay.sol:98]: Should be done in the latest version!

File: contracts/platform/__tests__/pay/processor.test.ts (hunk 5 above)
Comment (end-r) [reply contracts/platform/__tests__/pay/processor.test.ts:50]: Should be done in the latest version!

File: contracts/platform/contracts/pay/Pay.sol (hunk 3 above)
Comment (kermankohli) [reply contracts/platform/contracts/pay/Pay.sol:98]: It still show it's all on the same line but maybe something wrong on my end - will leave to you tho

**Pull request merged**